README.md
# FTweaks
Various options and tweaks.
-----------------
Primary Features:
Option for larger stack sizes (200) of base materials (ore, plates, circuits, etc)
Option for bobplates-style cheaper steel (2 plates and 50% faster)
Option to make the rocket silo and the rockets/satellites require far more research and infrastructure (includes BobMod integration)
Option to gate nuclear reactors behind far more research
Option to make rocket launches greatly upset the enemies, triggering large attacks and evolution factor increase
Makes loader technologies require each previous tier if present
Increases max range for green/purple Bob belts if present
Increases green (Tier 3) assembler from 1.25x to 1.5x speed, as if the four speed modules inside actually worked
Changes spawners' damage resistance to explosions from 5/15% to negative, as big bubbly things are not known for shockwave resilience, and there is otherwise no realistic way to destroy spawners pre-oil
KSPower windmills are made somewhat cheaper
Big Brother expanded-area radars now scan further but only monitor the normal area


Secondary Features:
Automatically reloads recipes and technologies when the game is reloaded, to keep up to date with mod changes
Small/Medium Powerpoles can now fast-replace each other
Lamps and rail signals use basic circuits instead of ones with components if BobElectronics is installed
Fixes some fluid-in-pipe colors, especially for the three oil products, to match the icons
Makes flight technology slightly harder and more sensible
Electric Energy II (Substations) needs Advanced Electronics I
Increases max speed for top-tier Bob Locomotives if present
Makes BobElectronics solder faster to craft if present
Adds a few conversion recipes for crafting low-tier machines into high-tier ones
Construction robots are now fire-resistant to prevent mass death in forest fires
Steam Turbine and Uranium Centrifuge now have their own sounds instead of sounding like steam engines and assemblers
Fluid Spills (mod) no longer prevent building placement
Accumulators are now somewhat quieter
Walking through water (if enabled by another mod) now makes splashing sounds

changelog.txt
---------------------------------------------------------------------------------------------------
Version: 1.101.148
Date: Oct 31 2022
  Misc Changes:
    - BobMod integration: Added missing module contact to tier 3 of all module colors
    - BobMod integration: Fixed offset of charge pads
    - BobMod integration: Increased filter slot count of high-tier filter inserters, 3 for blue and 5 for purple
    - More fixes to normal/expensive split in red science split option
---------------------------------------------------------------------------------------------------
Version: 1.101.144
Date: Oct 7 2022
  Bugfixes:
    - BobMod integration: Fixed crash with solar equipment
    - Fixed normal/expensive split in red science split option
  New Features:
    - Added "identify" command to help in debugging
    - Added option to cheapen landfill
  Rebalancing:
    - Increased output temperature of KSPower steam boiler to be comparable to a BobMods boiler 2
  Misc Changes:
    - BobMod integration: Added option to reduce gem cutting efficiency
    - BobMod integration: Increased lead cost of battery
    - BobMod integration: Increased sodium hydroxide cost of blue science
    - BobMod integration: Removed collision from robot charge pads
    - Updated schall pickup tower recipe changes
  Removed Features:
    - Removed cobalt steel from blue belt recipe
    - Removed train-signal collision removal
---------------------------------------------------------------------------------------------------
Version: 1.101.105
Date: Apr 28 2021
  Misc Changes:
    - Update to 1.1
---------------------------------------------------------------------------------------------------
Version: 1.018.24
Date: Nov 15 2020
  New Features:
    - Added handling for AAI industry for early upcrafting recipes and armor tiering
  Rebalancing:
    - Increased item collector equipment range from 40 to 60
  Removed Features:
    - Removed Schall changes
---------------------------------------------------------------------------------------------------
Version: 1.018.18
Date: Aug 10 2020
  New Features:
    - Added option and function to increase artillery range and (if EGCombat is not installed) damage
    - Added optional Item Collector equipment
  Misc Changes:
    - Advanced satellite now requires space science, not just rocket silo
  Removed Features:
    - Removed now-native changes to Bob Electrolyzer 3 (tungsten vs aluminum)
---------------------------------------------------------------------------------------------------
Version: 1.018.15
Date: Jul 28 2020
  New Features:
    - Added code to prevent inserters from interacting with Helicopters, preventing the need for its "disable nearby inserters" option
    - Added option and function to make cliff explosives cheaper
    - Added upcrafting recipe from steam engine to steam turbine
---------------------------------------------------------------------------------------------------
Version: 1.018.6
Date: Jun 15 2020
  New Features:
    - Added code to rebalance Schall Pickup towers and make them sequentially gated
    - Added command to reroute all trains using a particular stop
    - Added option and function to make Construction Drones invincible
---------------------------------------------------------------------------------------------------
Version: 1.018.5
Date: Jun 11 2020
  New Features:
    - Added option and function for steam cracking (ie use of steam rather than water in cracking recipes)
    - Added option and function for tech demultiply on upgrade-type techs
  Misc Changes:
    - Edited some tech prereqs around electronics and plastic
---------------------------------------------------------------------------------------------------
Version: 1.018.2
Date: Jun 7 2020
  New Features:
    - Added "Automation" tech prereq to bob greenhouses
    - Added code to change Construction Drone collision masks
  Misc Changes:
    - Changed power usage of Electric Offshore Pump
  Removed Features:
    - Removed rubberbelts crappy belt upcrafting, as that is now native
---------------------------------------------------------------------------------------------------
Version: 1.018.1
Date: May 29 2020
  Bugfixes:
    - Fixed some compat issues in advanced satellite recipe
  Misc Changes:
    - Update to 0.18
---------------------------------------------------------------------------------------------------
Version: 1.017.35
Date: Dec 29 2019
  New Features:
    - Added code to make all chemplant recipes have colorized exhaust if they do not already, calculated based on the colors of the fluids
---------------------------------------------------------------------------------------------------
Version: 1.017.24
Date: Sep 30 2019
  Bugfixes:
    - Fixed some crashes on load where a non-bob mod adds some bob materials but not others
  New Features:
    - Added option to add iron pipe to green science like the abandoned vanilla idea from FFF 275
---------------------------------------------------------------------------------------------------
Version: 1.017.15
Date: Jul 1 2019
  Bugfixes:
    - Fixed handling of bob steam power and belts
---------------------------------------------------------------------------------------------------
Version: 1.017.12
Date: Jun 17 2019
  Misc Changes:
    - More 0.17 fixes
---------------------------------------------------------------------------------------------------
Version: 1.017.6
Date: Jun 6 2019
  Bugfixes:
    - Fixed braking power modification for Helicopters
    - Fixed recipe handling errors
  Misc Changes:
    - Moved shared code and recipe library to Dragon Industries
    - Update to 0.17
  Removed Features:
    - Removed rocketry dependency from Helicopters
---------------------------------------------------------------------------------------------------
Version: 1.016.180
Date: Nov 28 2018
  Bugfixes:
    - Fixed commands not being accessible the first load of a save

prototypes/wood-sludge.lua
if Config.woodSludge then

data:extend({
  {
    type = "recipe",
    name = "wood-sludge",
    category = "oil-processing",
    enabled = false,
    energy_required = 4,
    ingredients =
    {
      {type="item", name="wood", amount=20},
      {type="fluid", name="sulfuric-acid", amount=20},
      {type="fluid", name="steam", amount=40}
    },
    results=
    {
      {type="fluid", name="heavy-oil", amount=10},
      {type="fluid", name="light-oil", amount=30},
      --{type="item", name="coal", amount=1}
    },
    icon = "__FTweaks__/graphics/icons/wood-sludge.png",
	icon_size = 32,
    subgroup = "fluid-recipes",
    order = "a[oil-processing]-c[wood-sludge]",
    allow_decomposition = false
  },
  {
    type = "recipe",
    name = "wood-sludge",
    category = "oil-processing",
    enabled = false,
    energy_required = 9,
    ingredients =
    {
      {type="item", name="wood", amount=10},
      {type="fluid", name="sulfuric-acid", amount=25},
      {type="fluid", name="steam", amount=50}
    },
    results=
    {
      {type="fluid", name="heavy-oil", amount=15},
      {type="fluid", name="light-oil", amount=35},
      --{type="item", name="coal", amount=1}
    },
    icon = "__FTweaks__/graphics/icons/wood-sludge.png",
	icon_size = 32,
    subgroup = "fluid-recipes",
    order = "a[oil-processing]-c[wood-sludge]",
    allow_decomposition = false
  },
  {
    type = "technology",
    name = "wood-sludge",
    icon = "__base__/graphics/technology/coal-liquefaction.png",
	icon_size = 128,
    prerequisites = {
		"coal-liquefaction",
		"advanced-material-processing-2",
		"sulfur-processing",
	},
    effects =
    {
      {
        type = "unlock-recipe",
        recipe = "wood-sludge"
      },
      {
        type = "unlock-recipe",
        recipe = "wood-sludge"
      }
    },
    unit =
    {
      count = 300,
      ingredients =
      {
        {"automation-science-pack", 1},
        {"logistic-science-pack", 1},
        {"chemical-science-pack", 1},
        {"production-science-pack", 1}
      },
      time = 30
    },
    order = "d-c"
  },
})

end